### PR TITLE
[WIP] Removed all the data and configs

### PR DESCRIPTION
**Pull request by sadasant**
Currently re-doing it.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>buildAgg should work</summary>
  <code>expected { Object (key) } to deeply equal { Object (key) }</code>
</details>
<details>
  <summary>buildFilter should work</summary>
  <code>expected { Object (type) } to deeply equal { type: { field: { someData: true } } }</code>
</details>
<details>
  <summary>should work</summary>
  <code>expected { Object (aggs) } to deeply equal { Object (aggs) }</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :exclamation: This PR is BIG (+745 -999)  
Please keep it below 500 net changes
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  You reduced the total lines of code! Awesome! :+1:
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/example-types/groupedMetric.js [hunk 1]
```diff
@@ -42,15 +38,15 @@ module.exports = {
                     : {
                         field: agg.field,
                       },
-                  agg.data
+                  agg
```
Comment (daedalus28) [inline src/example-types/groupedMetric.js]: I don't think this is right because it is inside `agg`, which is in the aggs array.

File: src/example-types/nLevelAggregation.js [hunk 2]
```diff
@@ -64,30 +64,23 @@ module.exports = {
             {
               field: agg.field,
             },
-            agg.data
+            agg
```
Comment (daedalus28) [inline src/example-types/nLevelAggregation.js]: again i think this one is wrong

File: test/example-types/groupedMetric.js [hunk 3]
```diff
@@ -56,6 +46,7 @@ describe('groupedMetric', () => {
         aggs: {
           child: {
             terms: {
+              type: 'terms',
```
Comment (daedalus28) [inline test/example-types/groupedMetric.js]: why did you add this property?

File: test/example-types/groupedMetric.js [hunk 4]
```diff
@@ -65,10 +56,12 @@ describe('groupedMetric', () => {
                   child: {
                     sum: {
                       field: 'PO.IssuedAmount',
+                      type: 'sum',
```
Comment (daedalus28) [inline test/example-types/groupedMetric.js]: ?

File: src/aggUtils.js [hunk 5]
```diff
@@ -2,14 +2,14 @@ let _ = require('lodash/fp')
 module.exports = {
   buildAgg: agg => ({
     [agg.key || agg.type]: {
-      [agg.type]: _.extend(agg.data, {
+      [agg.type]: _.extend(agg, {
```
Comment (daedalus28) [inline src/aggUtils.js]: be careful with this - `agg` is not necessarily a context node

File: src/example-types/esTwoLevelAggregation.js [hunk 6]
```diff
@@ -78,10 +78,10 @@ module.exports = {
             .twoLevelAgg.buckets
         ),
       }
-      if (context.config.extraAggs) {
+      if (context.extraAggs) {
         _.each(agg => {
-          rtn[agg.key] = results.aggregations[agg.key][agg.config.value_field]
-        }, context.config.extraAggs)
+          rtn[agg.key] = results.aggregations[agg.key][agg.value_field]
```
Comment (daedalus28) [inline src/example-types/esTwoLevelAggregation.js]: not sure if we should be changing the API for the contents of extraAggs

File: test/example-types/groupedMetric.js (hunk 4 above)
Comment (sadasant) [reply test/example-types/groupedMetric.js]: Read the description of the PR
